### PR TITLE
Update microservices samples to sbt 1.6.1

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: docs-source/docs/modules/microservices-tutorial/examples/00-shopping-analytics-service-scala/build.sbt [hunk 1]
```diff
@@ -19,7 +19,19 @@ Test / parallelExecution := false
 Test / testOptions += Tests.Argument("-oDF")
 Test / logBuffered := false
 
-run / fork := false
+run / fork := true
+run / javaOptions :=
```
Comment (raboof) [inline docs-source/docs/modules/microservices-tutorial/examples/00-shopping-analytics-service-scala/build.sbt:23]: it's a bit unfortunate that we need to 'channel' any system properties from sbt to the forked JVM, but the good thing is it gives us an opportunity to give a proper error message when you don't select a configuration, which wouldn't work for these samples.

File: docs-source/docs/modules/microservices-tutorial/examples/00-shopping-cart-service-scala/build.sbt [hunk 2]
```diff
@@ -19,7 +19,19 @@ Test / parallelExecution := false
...
 Test / logBuffered := false
 
-run / fork := false
+run / fork := true
+run / javaOptions :=
+  javaOptions.value ++ (
+    sys.props.get("config.resource") match {
+      case None =>
+        throw new IllegalStateException(
+          "Please specify a configuration to use, e.g. " ++
```
Comment (patriknw) [inline docs-source/docs/modules/microservices-tutorial/examples/00-shopping-cart-service-scala/build.sbt:28]: shouldn't this be `+` for ordinary Java string concatenation?

File: docs-source/docs/modules/microservices-tutorial/examples/00-shopping-cart-service-scala/build.sbt [hunk 3]
```diff
@@ -19,7 +19,19 @@ Test / parallelExecution := false
 Test / testOptions += Tests.Argument("-oDF")
 Test / logBuffered := false
 
-run / fork := false
+run / fork := true
+run / javaOptions :=
+  javaOptions.value ++ (
+    sys.props.get("config.resource") match {
+      case None =>
+        throw new IllegalStateException(
```
Comment (patriknw) [inline docs-source/docs/modules/microservices-tutorial/examples/00-shopping-cart-service-scala/build.sbt:27]: IllegalArgumentException instead?

File: docs-source/docs/modules/microservices-tutorial/examples/00-shopping-analytics-service-scala/build.sbt [hunk 4]
```diff
@@ -19,7 +19,19 @@ Test / parallelExecution := false
...
+    sys.props.get("config.resource") match {
+      case None =>
+        throw new IllegalArgumentException(
+          "Please specify a configuration to use, e.g. " +
+          " `sbt -Dconfig.resource=local1.conf run`"
+        )
+      case Some(r) =>
+        Seq(s"-Dconfig.resource=$r")
+    }
+  )
```
Comment (octonato) [inline docs-source/docs/modules/microservices-tutorial/examples/00-shopping-analytics-service-scala/build.sbt:34]: And what if we default to local1.conf if none is passed?

**Pull request closed without merging**